lib/daru/category.rb

```diff
@@ -922,6 +922,23 @@ def modify_category_at pos, category
       @cat_hash[category] << pos
     end
 
+    def set pos, category
+      unless categories.include? category
+        raise ArgumentError, "Invalid category #{category}, "\
+          'to add a new category use #add_category'
+      end
+      pos = pos.is_a?(Array) ? pos : [pos]
+      pos.each do |x|
```

Comment:
Please make sure that this would also work for iterators like dv.set 1..10, 7

Also it would be good if you name your variables depending upon what purpose they serve. For example x can be replaced by pos and pos by positions

Reply:
I think you can look at vector.rb on how to do that.

Reply:
please review